frontend/src/components/utils/store/index.js

```diff
@@ -0,0 +1,21 @@
+import { create } from 'zustand'
+
+const availableStores = {}
+
+export function createGlobalStore(name, initialState) {
+    const existingStore = availableStores[name];
```

Comment:
@Agaba-derrick am not sure what you are trying to do here but looks like you're making this unnecessarily complicated and I think it should be simple